src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +33,159 @@ const ActorInit Dm_Hina_InitVars = {
+        this->actionFunc = DmHina_SetupSubCamera;
+    }
+}
+
+void DmHina_SetupSubCamera(DmHina* this, PlayState* play) {
+    this->timer--;
+    if (this->timer == 0) {
+        this->isVisible = true;
+        Cutscene_Start(play, &play->csCtx);
+        this->subCamId = Play_CreateSubCamera(play);
+        Play_CameraChangeStatus(play, CAM_ID_MAIN, CAM_STATUS_WAIT);
+        Play_CameraChangeStatus(play, this->subCamId, CAM_STATUS_ACTIVE);
+        this->actionFunc = DmHina_MoveSubCamera;
+    }
+}
+
+void DmHina_MoveSubCamera(DmHina* this, PlayState* play) {
+    this->subCamEye.x = this->actor.world.pos.x + 100.0f;
+    this->subCamEye.y = this->blueWarpPosY + 260.0f;
+    this->subCamEye.z = this->actor.world.pos.z + 100.0f;
+    this->subCamAt.x = this->actor.world.pos.x;
+    this->subCamAt.y = this->actor.world.pos.y + (this->unk_154 * this->unk_15C) + (40.0f * this->unk_15C);
+    this->subCamAt.z = this->actor.world.pos.z;
+    Play_CameraSetAtEye(play, this->subCamId, &this->subCamAt, &this->subCamEye);
+    Math_SmoothStepToF(&this->actor.world.pos.y, this->blueWarpPosY + 300.0f, 0.5f, 2.0f, 0.1f);
+    if (((this->blueWarpPosY + 240.0f) < this->actor.world.pos.y) && (this->csState != 1)) {
+        this->csState = 1;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_OC_WHITE_OUT_INTO_KYOJIN);
+    }
+    Actor_PlaySfxAtPos(&this->actor, 0x219B);
+}
+
+void func_80A1F75C(DmHina* this, PlayState* play) {
+    s32 i;
+    s16 light1Color;
+
+    switch (this->csState) {
+        case 0:
+            this->lightOrbAlpha = 0;
+            this->scale = 0.0f;
+            break;
+
+        case 1:
+            Math_SmoothStepToF(&this->scale, 1.0f, 0.4f, 0.05f, 0.001f);
+            this->lightOrbAlpha = this->scale * 255.0f;
+
+            this->unk_150 = Math_SinS(play->state.frames * 0x1F40);
+
+            for (i = 0; i < 3; i++) {
+                light1Color = this->scale * -255.0f;
+                play->envCtx.lightSettings.diffuseColor1[i] = light1Color;
+                play->envCtx.lightSettings.fogColor[i] = light1Color;
+                play->envCtx.lightSettings.ambientColor[i] = light1Color;
+            }
+
+            play->envCtx.lightSettings.fogNear = this->scale * -500.0f;
+
+            if (play->envCtx.lightSettings.fogNear < -300) {
+                play->roomCtx.currRoom.segment = NULL;
+            }
+
+            break;
+    }
+}
+
+void DmHina_Update(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->actionFunc(this, play);
+    func_80A1F75C(this, play);
+}
+
+void func_80A1F9AC(DmHina* this, PlayState* play) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+
+    if (this->csState != 0) {
+        OPEN_DISPS(gfxCtx);
+
+        func_8012C2DC(play->state.gfxCtx);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, (u8)(this->scale * 100.0f) + 155, this->lightOrbAlpha);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, (u8)(this->unk_150 * 100.0f) + 50, 0);
```

Comment:
Looks like unk150 is going to be the blue color value of env?

`envColorB` maybe?